include/flatbuffers/util_base64.h

```diff
@@ -0,0 +1,64 @@
+#ifndef FLATBUFFERS_UTIL_B64_H_
+#define FLATBUFFERS_UTIL_B64_H_
+
+namespace flatbuffers {
+// [BASE64]{https://tools.ietf.org/html/rfc4648}
+
+// a Field doesn't have base64/base64url attribute
+static const int kBase64Undefined = 0;
+
+// attribute (base64): standard RFC4648 alphabet
+// decoder: padding is optional
+// encoder: with padding
+static const int kBase64Standard = 1;
+
+// attribute (base64url): RFC4648 with URL and Filename Safe Alphabet
+// decoder: padding is optional
+// encoder: padding can be canceled using kBase64CancelPadding flag
+static const int kBase64UrlSafe = 2;
+
+// Cancel padding for Base64Url encoder
+static const int kBase64CancelPadding = 4;
+
+// Helper: FDT is <idl.h>::FieldDef type
+template<typename FDT> inline int field_base64_mode(const FDT *fd) {
```

Comment:
Just make it rely on `FieldDef` directly rather than this template trick. You can include `idl.h` since this this file is included in places where `idl.h` was already included anyway.